src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/correspondence/generalorder/FinremGeneralOrderRaisedCorresponder.java

```diff
@@ -0,0 +1,74 @@
+package uk.gov.hmcts.reform.finrem.caseorchestration.service.correspondence.generalorder;
+
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.FinremCaseDetails;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.wrapper.intevener.IntervenerWrapper;
+import uk.gov.hmcts.reform.finrem.caseorchestration.model.wrapper.SolicitorCaseDataKeysWrapper;
+import uk.gov.hmcts.reform.finrem.caseorchestration.service.CaseDataService;
+import uk.gov.hmcts.reform.finrem.caseorchestration.service.NotificationService;
+import uk.gov.hmcts.reform.finrem.caseorchestration.service.correspondence.FinremEmailOnlyAllSolicitorsCorresponder;
+
+@Component
+public class FinremGeneralOrderRaisedCorresponder extends FinremEmailOnlyAllSolicitorsCorresponder {
+
+    private final CaseDataService caseDataService;
+
+    public FinremGeneralOrderRaisedCorresponder(NotificationService notificationService,
+                                                CaseDataService caseDataService) {
+        super(notificationService);
+        this.caseDataService = caseDataService;
+    }
+
+    public void sendCorrespondence(FinremCaseDetails caseDetails) {
+        super.sendCorrespondence(caseDetails);
+        sendIntervenerCorrespondence(caseDetails);
+    }
+
+    private void sendIntervenerCorrespondence(FinremCaseDetails caseDetails) {
+        if (!caseDataService.isContestedApplication(caseDetails)) {
+            return;
+        }
+
+        caseDetails.getData().getInterveners().stream()
+            .filter(i -> isEmailToIntervenerSolicitorRequired(caseDetails, i))
+            .forEach(i -> emailIntervenerSolicitor(caseDetails, i));
+    }
+
+    @Override
+    protected void emailApplicantSolicitor(FinremCaseDetails caseDetails) {
+        if (caseDataService.isConsentedApplication(caseDetails)) {
+            notificationService.sendConsentedGeneralOrderEmailToApplicantSolicitor(caseDetails);
+        } else {
+            if (caseDataService.isConsentedInContestedCase(caseDetails)) {
```

Comment:
This method checking only D81 not event? Please check if this wouldn't create problem

Reply:
Yes agreed this doesn't seem to match the new general order logic. Following our discussion yesterday I'll update this and Luciano will recreate this issue

Reply:
I've updated this check